### PR TITLE
Run python-black over the source tree.

### PR DESCRIPTION
**Pull request by jameinel**
Mostly this just wraps long lines and changes '' strings into ""
strings. The big reason to do this is that we can commit to it like
people do for "go fmt". It is the "One True Formatting" that means we
don't have to worry about it, and don't have to end up fixing flake8
warnings. You can configure an editor to automatically run black on your
source file, and that will allow you to not have to worry about
formatting again.

See https://black.readthedocs.io/en/stable/

This is mostly because I wanted to use black, but I didn't want a small
change of mine to get drowned in lots of automatic changes. If we
decide that we don't want black, I can revert this and reconfigure my editor.

### CODE REVIEW

**Comment by johnsca** [comment]
This is now failing on flake8.  It surprises me that python-black and flake8 are arguing.

**Comment by jameinel** [comment]
Yeah, I saw that. I looked at the error and it is that python-black reformats deeply nested structs as:
```
"stor-4": {"type": "filesystem", "multiple": {"range": "2-4",},},
```
And it seems flake8 wants
```
"stor-4": {"type": "filesystem", "multiple": {"range": "2-4", }, },
```

**Comment by johnsca** [comment]
flake8's requirement there is dumb.  :)  +1 to switching to `black -c`

**Comment by jameinel** [comment]
Supporting .gitignore is a known issue in black: https://github.com/psf/black/issues/475
It seems it is "lots of people think it would be good" but it hasn't been implemented.

For now, I added the exclude config.

File: test/test_framework.py [hunk 1]
```diff
@@ -557,131 +570,80 @@ def test_mutable_types(self):
...
+            (set(["a"]), "a", set(), lambda a, b: a.discard(b), lambda res, expected_res: self.assertEqual(res, expected_res),),
+            (
+                set(),
+                {"a"},
+                set(),
+                # Nested sets are not allowed as sets themselves are not hashable.
+                lambda a, b: self.assertRaises(TypeError, a.add, b),
+                lambda res, expected_res: self.assertEqual(res, expected_res),
+            ),
+        ]
```
Comment (niemeyer) [inline test/test_framework.py]: This one looks much worse.

File: test/test_framework.py [hunk 2]
```diff
@@ -691,93 +653,41 @@ class SomeObject(Object):
...
+            ({"a", "c"}, {"c", "a"}, lambda a, b: a == b, True, True),
+            (dict(), dict(), lambda a, b: a == b, True, True),
+            ({"1": "2"}, {"1": "2"}, lambda a, b: a == b, True, True),
+            ({"1": "2"}, {"1": "3"}, lambda a, b: a == b, False, False),
+            ([], [], lambda a, b: a == b, True, True),
+            ([1, 2], [1, 2], lambda a, b: a == b, True, True),
+            ([1, 2, 5, 6], [1, 2, 5, 8, 10], lambda a, b: a <= b, True, False),
+            ([1, 2, 5, 6], [1, 2, 5, 8, 10], lambda a, b: a < b, True, False),
+            ([1, 2, 5, 8], [1, 2, 5, 6, 10], lambda a, b: a > b, True, False),
+            ([1, 2, 5, 8], [1, 2, 5, 6, 10], lambda a, b: a >= b, True, False),
```
Comment (niemeyer) [inline test/test_framework.py]: Same.. the test was organized with readability in mind, and it all looks reasonable. It completely ignored the choice of formatting and blind-reformatted it based only on line length.

File: test/test_main.py [hunk 3]
```diff
@@ -80,177 +81,177 @@ def tearDown(self):
...
-            if 'broken' not in event_name:
-                env.update({
-                    'JUJU_REMOTE_UNIT': 'remote/0',
-                })
+        if "relation" in event_name:
+            rel_name = event_name.split("_")[0]
+            rel_id = {"db": "1", "mon": "2", "ha": "3"}[rel_name]
+            env.update(
+                {"JUJU_RELATION": rel_name, "JUJU_RELATION_ID": rel_id,}
+            )
```
Comment (niemeyer) [inline test/test_main.py]: Also looks worse.. the left-hand side feels very nicely formatted.

**Comment by niemeyer** [comment]
We agreed to not pursue this for now. Besides the bug, Black is reformatting code even when it looks good according to its own standard, just because it fits in the same line and could be line-merged. We need to either find a different tool which respects proper style slightly more.

**Pull request closed without merging**